yt_dlp/extractor/japandiet.py

```diff
@@ -0,0 +1,281 @@
+import re
+
+from ..utils import (
+    ExtractorError,
+    clean_html,
+    int_or_none,
+    join_nonempty,
+    parse_qs,
+    smuggle_url,
+    traverse_obj,
+    try_call,
+    unsmuggle_url
+)
+from .common import InfoExtractor
+
+
+class ShugiinItvBaseIE(InfoExtractor):
+    @classmethod
+    def _find_rooms(cls, webpage):
+        return [{
+            '_type': 'url',
+            'id': x.group(1),
+            'title': clean_html(x.group(2)).strip(),
+            'url': smuggle_url(f'https://www.shugiintv.go.jp/jp/index.php?room_id={x.group(1)}', {'g': x.groups()}),
+            'ie_key': ShugiinItvLiveIE.ie_key(),
+        } for x in re.finditer(r'(?s)<a\s+href="[^"]+\?room_id=(room\d+)"\s*class="play_live".+?class="s12_14">(.+?)</td>', webpage)]
+
+    @staticmethod
+    def _parse_japanese_date(text):
+        if not text:
+            return None
+        ERA_TABLE = {
+            '明治': 1868,
+            '大正': 1912,
+            '昭和': 1926,
+            '平成': 1989,
+            '令和': 2019,
+        }
+        ERA_RE = '|'.join(map(re.escape, ERA_TABLE.keys()))
+        mobj = re.search(rf'({ERA_RE})?(\d+)年(\d+)月(\d+)日', re.sub(r'[\s\u3000]+', '', text))
+        if not mobj:
+            return None
+        era, year, month, day = mobj.groups()
+        year, month, day = map(int, (year, month, day))
+        if era:
+            # example input: 令和5年3月34日
+            # even though each era have their end, don't check here
+            year += ERA_TABLE[era]
+        return '%04d%02d%02d' % (year, month, day)
+
+    @staticmethod
+    def _parse_japanese_duration(text):
+        if not text:
+            return None
+        mobj = re.search(r'(?:(\d+)日間?)?(?:(\d+)時間?)?(?:(\d+)分)?(?:(\d+)秒)?', re.sub(r'[\s\u3000]+', '', text))
+        if not mobj:
+            return None
+        days, hours, mins, secs = map(int_or_none, mobj.groups())
+
+        duration = 0
+        if secs:
+            duration += float(secs)
+        if mins:
+            duration += float(mins) * 60
+        if hours:
+            duration += float(hours) * 60 * 60
+        if days:
+            duration += float(days) * 24 * 60 * 60
+        return duration
+
+
+class ShugiinItvLiveIE(ShugiinItvBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?shugiintv\.go\.jp/(?:jp|en)(?:/index\.php)?$'
+    IE_DESC = '衆議院インターネット審議中継'
+
+    _TESTS = [{
+        'url': 'https://www.shugiintv.go.jp/jp/index.php',
+        # expect at least one proceedings is running
+        'playlist_mincount': 1,
+        'only_matching': True,
```

Comment:
why only_matching? The playlist metadata should be static

Reply:
added